aiven/resource_flink_test.go

```diff
@@ -0,0 +1,240 @@
+package aiven
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccAiven_flinkBasic(t *testing.T) {
+	resourceName := "aiven_flink.bar"
+	serviceName := fmt.Sprintf("test-acc-flink-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	projectName := os.Getenv("AIVEN_PROJECT_NAME")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAivenServiceResourceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
```

Comment:
Would you mind wrapping the config to a helper function? It is tough to read what is happening.

Reply:
:heavy_check_mark:  moved to a constant in the test